docs/modules/hdfs/pages/usage-guide/operations/pod-disruptions.adoc

```diff
@@ -0,0 +1,105 @@
+
+= Allowed Pod disruptions
+
+You can configure the allowed Pod disruptions for HDFS nodes as described in xref:concepts:operations/pod_disruptions.adoc[].
+
+Unless you configure something else or disable our PodDisruptionBudgets (PDBs), we write the following PDBs:
+
+== JournalNodes
+We only allow a single JournalNode to be offline at any given time, regardless of the number of replicas or `roleGroups`.
+
+== NameNodes
+We only allow a single NameNode to be offline at any given time, regardless of the number of replicas or `roleGroups`.
+
+== DataNodes
+For DataNodes the question how many instances can be unavailable at the same time is a bit harder:
+HDFS stores your blocks on the datanodes.
+Every block can be replicated multiple times (to multiple datanodes) to ensure maximum availability.
+The default replication factor is `3` - which can be configured using `spec.clusterConfig.dfsReplication`. However, it is also possible to change the replication factor for a specific file or directory to something other than the cluster default.
+
+When you have a replication of `3`, you can safely can take down 2 DataNodes, as there will always be a third DataNode holding a copy of each block currently assigned to one of the unavailable DataNodes.
+However, you need to be aware that you are now down to a single point of failure - the last of three replicas!
+
+Taking this into consideration, our operator uses the following algorithm to determine the maximum number of DataNodes allowed to be unavailable at the same time:
+
+`num_datanodes` is the number of datanodes in the HDFS cluster, summed over all `roleGroups`.
+
+`dfs_replication` is default replication factor of the cluster.
+
+[source,rust]
+----
+// There must always be a datanode to serve the block.
+// If we would simply subtract one from the `dfs_replication`, we would end up
+// with a single point of failure, so we subtract two instead.
+let max_unavailable = dfs_replication.saturating_sub(2);
+// We need to make sure at least one datanode remains by having at most
+// n - 1 datanodes unavailable. We subtract two to avoid a single point of failure.
+let max_unavailable = min(max_unavailable, num_datanodes.saturating_sub(2));
+// Clamp to at least a single node allowed to being, to not block Kubernetes nodes from draining.
```

Comment:
Hard to read and language tool complains as well :)

Reply:
Changed to "Clamp to at least a single node allowed to be offline, so we don't block Kubernetes nodes from draining."